python/packages/core/agent_framework/_types.py

```diff
@@ -1970,6 +1970,15 @@ def text(self) -> str:
         """
         return " ".join(content.text for content in self.contents if isinstance(content, TextContent))
 
+    @property
+    def reasoning(self) -> str:
```

Comment:
please don't add things without first discussing why we need this, not saying we won't need this, but we don't want to just add stuff for everything.